src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -599,4 +666,604 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
+      &appdomain_get_current_domain_member_ref);
+  delete[] appdomain_get_current_domain_signature;
+
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineMemberRef failed");
+    return hr;
+  }
+
+  // Create method signature for AppDomain.Load(byte[], byte[])
+  COR_SIGNATURE appdomain_load_signature_start[] = {
+      IMAGE_CEE_CS_CALLCONV_HASTHIS,
+      2,
+      ELEMENT_TYPE_CLASS  // ret = System.Reflection.Assembly
+      // insert compressed token for System.Reflection.Assembly TypeRef here
+  };
+  COR_SIGNATURE appdomain_load_signature_end[] = {
+      ELEMENT_TYPE_SZARRAY,
+      ELEMENT_TYPE_U1,
+      ELEMENT_TYPE_SZARRAY,
+      ELEMENT_TYPE_U1
+  };
+  start_length = sizeof(appdomain_load_signature_start);
+  auto end_length = sizeof(appdomain_load_signature_end);
+
+  BYTE system_reflection_assembly_type_ref_compressed_token[4];
+  token_length = CorSigCompressToken(system_reflection_assembly_type_ref, system_reflection_assembly_type_ref_compressed_token);
+
+  COR_SIGNATURE* appdomain_load_signature = new COR_SIGNATURE[start_length + token_length + end_length];
+  memcpy(appdomain_load_signature,
+         appdomain_load_signature_start,
+         start_length);
+  memcpy(&appdomain_load_signature[start_length],
+         system_reflection_assembly_type_ref_compressed_token,
+         token_length);
+  memcpy(&appdomain_load_signature[start_length + token_length],
+         appdomain_load_signature_end,
+         end_length);
+
+  mdMemberRef appdomain_load_member_ref;
+  hr = metadata_emit->DefineMemberRef(
+      system_appdomain_type_ref, "Load"_W.c_str(),
+      appdomain_load_signature,
+      start_length + token_length + end_length,
+      &appdomain_load_member_ref);
+  delete[] appdomain_load_signature;
+
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineMemberRef failed");
+    return hr;
+  }
+
+  // Create method signature for Assembly.CreateInstance(string)
+  COR_SIGNATURE assembly_create_instance_signature[] = {
+      IMAGE_CEE_CS_CALLCONV_HASTHIS,
+      1,
+      ELEMENT_TYPE_OBJECT,  // ret = System.Object
+      ELEMENT_TYPE_STRING
+  };
+
+  mdMemberRef assembly_create_instance_member_ref;
+  hr = metadata_emit->DefineMemberRef(
+      system_reflection_assembly_type_ref, "CreateInstance"_W.c_str(),
+      assembly_create_instance_signature,
+      sizeof(assembly_create_instance_signature),
+      &assembly_create_instance_member_ref);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineMemberRef failed");
+    return hr;
+  }
+
+  // Create a string representing "Datadog.Trace.ClrProfiler.Managed.Loader.Startup"
+  // Create OS-specific implementations because on Windows, creating the string via
+  // "Datadog.Trace.ClrProfiler.Managed.Loader.Startup"_W.c_str() does not create the
+  // proper string for CreateInstance to successfully call
+#ifdef _WIN32
+  LPCWSTR load_helper_str =
+      L"Datadog.Trace.ClrProfiler.Managed.Loader.Startup";
```

Comment:
Did you try this with `"some_string"_W`? That should work on both platforms. See `pal.h`.

Reply:
I had issues using `"string"_W` working on both Windows and Linux. Let me follow up on this when I get the Linux implementation up and running.